lib/large_object_store.rb

```diff
@@ -67,9 +70,9 @@ def read(key)
         # use values_at to enforce key order because read_multi doesn't guarantee a return order
         slices = @store.read_multi(*keys).values_at(*keys)
         return nil if slices.compact.size != pages
-        slices.map! { |s| [s.slice!(0, UUID_SIZE), s] }
+        slices = slices.map { |s| [s[MULTI_PAGE_UUID_RANGE], s[MULTI_PAGE_DATA_RANGE]] }
```

Comment:
still prefer the simplicity of `slice!` ... no code to be logic in sync and ...
`.map { |s| s = s.dup; [s.slice!(0, UUID_SIZE), s] } ` maybe ?